web/yo/app/scripts/directives/addtherapy.js

```diff
@@ -77,10 +77,15 @@ angular.module('oncokbApp')
 
                 $scope.loadDrugs = function($query) {
                     $scope.addTherapyError = false;
-                    return drugs.filter(function (drug) {
-                        var lowerCaseQuery = $query.toLowerCase();
-                        return drug.drugName.toLowerCase().indexOf(lowerCaseQuery) != -1 || (!drug.synonyms ? false : drug.synonyms.join(',').toLowerCase().indexOf(lowerCaseQuery) != -1);
-                    })
+                    var lowerCaseQuery = $query.toLowerCase();
+                    var result = [];
+                    result = drugs.filter(function (drug){
```

Comment:
Why twice?

Reply:
First: return drugs whose drug name contain the query.
Second: return drugs whose synonyms contain the query.
To make drug list in order.

Reply:
I would instead considering just introduce a sorting function. `.filter` is not for this purpose, and you also need to do a uniq and concat. Please also consider then a drug name starts with the query should have a higher priority then the drug name has the query.